src/component/DetectionComponent.ts

```diff
@@ -1,127 +1,109 @@
 /// <reference path="../../typings/browser.d.ts" />
 
 import * as rx from "rx";
+import * as vd from "virtual-dom";
 
 import {Node} from "../Graph";
 import {Container, Navigator} from "../Viewer";
-import {APIv2, IRect} from "../API";
+import {APIv3} from "../API";
 
 import {ComponentService, Component} from "../Component";
+import {IVNodeHash} from "../Render";
+
+interface IRect {
```

Comment:
This interface contains more information than just a rectangle. IDetection feels like a more informative name.